core/src/codegen/from_meta_impl.rs

```diff
@@ -90,16 +89,42 @@ impl<'a> ToTokens for FromMetaImpl<'a> {
                     }
                 };
 
+                let default_or_err = base
+                    .default
+                    .as_ref()
+                    .map(|default_expr| quote!(::darling::export::Ok(#default_expr)))
+                    .unwrap_or_else(|| {
+                        quote!(::darling::export::Err(::darling::Error::too_few_items(1)))
+                    });
+
+                let word_or_err = variants
+                    .iter()
+                    .find_map(|variant| {
+                        if variant.word {
+                            let ty_ident = variant.ty_ident;
+                            let variant_ident = variant.variant_ident;
+                            Some(quote!(::darling::export::Ok(#ty_ident::#variant_ident)))
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or_else(|| {
+                        quote!(::darling::export::Err(
+                            ::darling::Error::unsupported_format("word")
+                        ))
+                    });
+
                 quote!(
                     fn from_list(__outer: &[::darling::export::NestedMeta]) -> ::darling::Result<Self> {
                         // An enum must have exactly one value inside the parentheses if it's not a unit
-                        // match arm
+                        // match arm.
+                        // NOTE: The empty list case uses the annotated default value for the enum if any.
```

Comment:
I'm not sure I agree with this behavior. Looking at `derive_buider` as an example, we see a difference in behavior between the "absent" case and the word-only case. If we get a field name with no contents, I'd expect that to error unless there was a `#[darling(word)]` variant present.